src/Microsoft.TestPlatform.Client/Discovery/DiscoveryRequest.cs

```diff
@@ -53,6 +57,11 @@ void IDiscoveryRequest.DiscoverAsync()
                 this.discoveryInProgress = true;
                 try
                 {
+                    this.discoveryStartTime = DateTime.UtcNow;
+
+                    // Collecting Data Point Number of sources sent for discovery
+                    this.requestData.MetricsCollection.Add(TelemetryDataConstants.NumberOfSourcesSentForDiscovery, (this.DiscoveryCriteria.Sources.Count()).ToString());
```

Comment:
Telemetry collection should not cause any failures.
Ensure we have proper exception handling and logging.